app/forms/questionnaire_form.py

```diff
@@ -480,32 +480,47 @@ def get_answer_fields(
     if block_ids_by_section:
         block_ids = get_flattened_mapping_values(block_ids_by_section)
 
-    def _get_value_source_resolver(list_item: str | None = None) -> ValueSourceResolver:
+    def _get_value_source_resolver(
```

Comment:
If this private method does not use vars from the higher scope of the class implicitly (is only passing parameters explicitly) , there is no reason why to nest it inside another method of this class. Should be separated out from the parent method.

Reply:
I think it still uses the higher scope of the function. If not I think the function would not be needed